charts/partials/templates/_metadata.tpl

```diff
@@ -9,6 +9,7 @@ linkerd.io/created-by: {{ .Values.cliVersion | default (printf "linkerd/helm %s"
 {{- define "partials.proxy.annotations" -}}
 linkerd.io/proxy-version: {{.Values.proxy.image.version | default .Values.linkerdVersion}}
 cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
+linkerd.io/trust-root-checksum: {{ .Values.identityTrustAnchorsPEM | sha256sum }}
```

Comment:
In the issue I suggested using an annotation that reflects the signature algorithm so that it's not implicit (if you actually wanted to validate it, for instance in a check). This means that when we have to change the signature algorithm, we can do so gracefully.

```suggestion
linkerd.io/trust-root-sha256: {{ .Values.identityTrustAnchorsPEM | sha256sum }}
```